bundles/org.eclipse.text.quicksearch/src/org/eclipse/text/quicksearch/internal/core/QuickTextQuery.java

```diff
@@ -48,7 +48,7 @@ public int getOffset() {
 
 	private boolean caseInsensitive;
 	private String orgPattern; //Original pattern case preserved even if search is case insensitive.
-	private Pattern pattern;
+	final Pattern pattern;
```

Comment:
Looking from my iPhone I can’t tell if the not being private is intentional.

Reply:
it's intentional package private now to share it across threads

Reply:
Maybe use a getter instead? Is the pattern reqully required or not more the matcher?

So one can thing of `public Matcher createMatcher(String input);`

Reply:
Pattern is thread safe, while Matcher is not.  This works. Unless you come up with a good reason to change it i would keep it as is. As the pattern is final as in a record there is no benefit of hiding it behind a getter.